sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,330 @@
+
+    Organizations are stored in 'organizations' object. Its keys
+    are the name of the organizations and each organization object is
+    related to a list of domains.
+
+    :param stream: stream to parse
+
+    :raises InvalidFormatError: raised when the format of the stream is
+        not valid.
+    """
+
+    EMAIL_ADDRESS_REGEX = r"^(?P<email>[^\s@]+@[^\s@.]+\.[^\s@]+)$"
+
+    def __init__(self, identities=None, organizations=None,
+                 source='grimoirelab'):
+        self._identities = {}
+        self._organizations = {}
+        self.source = source
+
+        if not (identities or organizations):
+            raise ValueError('Null identities and organization streams')
+
+        self.__parse(identities, organizations)
+
+    @property
+    def identities(self):
+        uids = [u for u in self._identities.values()]
+        uids.sort(key=lambda u: u.uuid)
+        return uids
+
+    @property
+    def organizations(self):
+        orgs = [o for o in self._organizations.values()]
+        orgs.sort(key=lambda o: o.name)
+        return orgs
+
+    def __parse(self, identities_stream, organizations_stream):
+        """Parse GrimoireLab stream"""
+
+        if organizations_stream:
+            self.__parse_organizations(organizations_stream)
+
+        if identities_stream:
+            self.__parse_identities(identities_stream)
+
+    def __parse_identities(self, stream):
+        """Parse identities using GrimoireLab format.
+
+        The GrimoireLab identities format is a YAML document following a
+        schema similar to the example below. More information available
+        at https://github.com/bitergia/identities
+
+        - profile:
+            name: Vivek K.
+            is_bot: false
+          github:
+            - vsekhark
+          email:
+            - vivek@****.com
+          enrollments:
+            - organization: Community
+              start: 1900-01-01
+              end: 2100-01-01
+
+        :parse json: YAML object to parse
+
+        :raise InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        def __create_sh_identities(name, emails, yaml_entry):
+            """Create SH identities based on name, emails and backens data in yaml_entry"""
+
+            ids = []
+            ids.append(Identity(name=name, source=self.source))
+
+            # FIXME we should encourage our users to add email or usernames
+            # and if not returning at least a WARNING
+            if emails:
+                for m in emails:
+                    ids.append(Identity(email=m, source=self.source))
+
+            for pb in PERCEVAL_BACKENDS:
+                if pb not in yaml_entry:
+                    continue
+
+                for username in yaml_entry[pb]:
+                    identity = Identity(username=username, source=pb)
+                    ids.append(identity)
+
+            return ids
+
+        yaml = self.__load_yml(stream)
```

Comment:
Don't redefine `yaml` variable. It was previously used to import `yaml` module. You should give it a different name.